### PR TITLE
mod : Damage taken by shield is now damage that weapons deals

### PR DESCRIPTION
**Pull request by namidaka**

### CODE REVIEW

**Comment by verdie-g** [comment]
image

**Review by verdie-g — CHANGES_REQUESTED** [review]
Could you fix the PR name, I don't know what you mean.

**Review by verdie-g — CHANGES_REQUESTED** [review]
Could you fix the PR name, I don't know what you mean.

**Comment by namidaka** [comment]
> Could you fix the PR name, I don't know what you mean.

my bad i thought you were talking about the pr. It's fixed. Basically , all weapon were doing the same damage no matter if they had 50c or 5c. I fixed it

File: src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs [hunk 1]
```diff
@@ -49,14 +49,22 @@ public CrpgAgentApplyDamageModel(CrpgConstants constants)
...
             {
-                // this bonus is on top of the native x2 in MissionCombatMechanicsHelper
-                // so the final bonus is 3.5. We do this instead of nerfing the impact of shield skill so shield can stay virtually unbreakable against sword.
-                // it is the same logic as arrows not dealing a lot of damage to horse but spears dealing extra damage to horses
-                // As we want archer to fear cavs and cavs to fear spears, we want swords to fear shielders and shielders to fear axes.
-                finalDamage *= 2f;
+                finalDamage *=
+                    (collisionData.StrikeType == (int)StrikeType.Thrust)
+                        ? weapon.CurrentUsageItem.ThrustDamageFactor
+                        : weapon.CurrentUsageItem.SwingDamageFactor;
```
Comment (verdie-g) [inline src/Module.Server/Common/Models/CrpgAgentApplyDamageModel.cs:58]: Coud you comment how you find out about that part. It doesn't seem intuitive.

Also I think you should get a warning for the redundant paranthesis.

**Review by verdie-g — CHANGES_REQUESTED** [review]

**Review by verdie-g — APPROVED** [review]

**Pull request merged**